### PR TITLE
feat(contracts): Implement Equipment Rental Management System

### PR DESCRIPTION
**Pull request by Spycall**
- Add equipment registration with metadata and geolocation
- Implement rental lifecycle (create, activate, complete, cancel)
- Add maintenance status tracking and enforcement
- Include comprehensive test suite for all features
- Handle edge cases like back-to-back rentals
- Add price calculation with proper validation

Tests: ✅ All tests passing

# 🌾 Revolutionary Farmers Pull Request 🚜

Mark with an `x` all the checkboxes that apply (like `[x]`)

- [] Closes #
- [] Added tests (if necessary)
- [] Run tests
- [] Run formatting
- [] Evidence attached
- [] Commented the code

---

### 📌 Type of Change

- [ ] Documentation (updates to README, docs, or comments)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

---

## 📝 Changes description

---

## 📸 Evidence (A photo is required as evidence)

---

## ⏰ Time spent breakdown

---

## 🌌 Comments

---

Thank you for contributing to Revolutionary Farmers, we are glad that you have chosen us as your project of choice and we hope that you continue to contribute to this great project, so that together we can make our mark at the top!

## Summary by CodeRabbit

- **Chores**
  - Updated `.gitignore` to exclude `.qodo` files or directories.
- **New Features**
  - Introduced a comprehensive equipment rental contract supporting equipment registration, rental lifecycle management, maintenance tracking, and error handling.
- **Tests**
  - Added extensive unit tests for the equipment rental contract, covering equipment registration, rental processes, maintenance, cancellations, and edge cases.
  - Added numerous test snapshot files for product auction and purchase review contracts, capturing ledger and contract states across diverse scenarios.
- **Style**
  - Standardized numeric representations in purchase review contract test snapshots by converting 128-bit integers to 64-bit integers for improved clarity and consistency.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,2 @@
-/target
\ No newline at end of file
+/target
+.qodo
```
Comment (aguilar1x) [inline .gitignore:2]: leave the gitignore as it was

File: ContractsRevo/equipment-rental-contract/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+edition = "2021"
+
+[lib]
+crate-type = ["cdylib"]
+
+[dependencies]
+soroban-sdk = "20.0.0"
+
+[dev_dependencies]
+soroban-sdk = { version = "20.0.0", features = ["testutils"] }
```
Comment (aguilar1x) [inline ContractsRevo/equipment-rental-contract/Cargo.toml:13]: dev_dependencies` is deprecated, please use

[dependencies]
soroban-sdk = { workspace = true }

[dev-dependencies]
soroban-sdk = { workspace = true, features = ["testutils"] }

**Comment by Spycall** [comment]
Please Review... no conflict

**Comment by aguilar1x** [comment]
> Please Review... no conflict

I still can't review it because your issue hasn't been unblocked yet. Your task is to write the test for the "equipment-rental-contract", which hasn't been merged. So please read the instructions in the issue.
